docs/concepts/parallelism.rst

```diff
@@ -0,0 +1,583 @@
+            return ["prompts"]
+
+        def writes() -> List[str]:
+            return ["all_llm_outputs"]
+
+Then, to run the application:
+
+.. code-block:: python
+
+    app = (
+        ApplicationBuilder()
+        .with_action(
+            prompt_generator=generate_prompts, # not defined above, this writes to prompts
+            multi_prompt_test=TestMultiplePrompts(),
+        ).with_transitions(
+            ("prompt_generator", "multi_prompt_test"),
+        )
+        .build()
+    )
+
+
+Run different actions over the same state
+-----------------------------------------
+
+
+For case (2) (mapping actions over the same state) you implement the ``MapActions`` class, doing the following:
+
+- We define a regular action ``query_llm_action()`` using the ``@action`` decorator. This takes in a model parameter (which we're going to bind later)
+- We also create a subclass of ``MapActions`` named ``TestMultipleModels``, which must implement ``.reads()``, ``.writes()``, ``.actions()``, ``.state()``, and ``.reduce()``.
+    - ``.reads()`` / ``.writes()`` define the state value it can interact with, just like the ``@action`` decorator
+    - ``.actions()`` leverages the ``query_llm_action()`` previously defined, binding with the different models we want to test
+    - ``.state()`` can read value from State and produces the state to pass to the actions produced by ``actions()``. In this case, it updates the prompt state value that's read by ``query_llm_action()``.
+    - ``.reduce()`` receives multiple states, one per result of the ``.actions()`` call, where the llm_output value is set by ``query_llm_action()`` in ``.actions()``. Then, it must set all_llm_output as specified in the ``MapStates.writes()`` method.
+- We pass an instance of the ``TestMultipleModels`` class to the ``ApplicationBuilder``, which will run the action over the states we provide.
+
+.. code-block:: python
+
+    from burr.core import action, state
+    from burr.core.parallelism import MapActions, RunnableGraph
+    from typing import Callable, Generator, List
+
+    @action(reads=["prompt", "model"], writes=["llm_output"])
+    def query_llm(state: State, model: str) -> State:
+        # TODO -- implement _query_my_llm to call litellm or something
+        return state.update(llm_output=_query_my_llm(prompt=state["prompt"], model=model))
+
+    class TestMultipleModels(MapActions):
+
+        def actions(self, state: State) -> Generator[Action | Callable | RunnableGraph, None, None]:
+            # make sure to add a name to the action
+            # This is not necessary for subgraphs, as actions will already have names
+            for action in [
+                query_llm.bind(model="gpt-4").with_name("gpt_4_answer"),
+                query_llm.bind(model="o1").with_name("o1_answer"),
+                query_llm.bind(model="claude").with_name("claude_answer"),
+            ]
+                yield action
+
+        def state(self, state: State) -> State::
```

Comment:
nit: I think there's a colon `:` missing from the `for` and an extra colon at the end of the `def state(..)` :)

Reply:
Thanks! Good find. Will update 🙏